src/main/java/io/mosip/digitalcard/service/impl/DigitalCardServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+package io.mosip.digitalcard.service.impl;
+
+import io.mosip.digitalcard.constant.DigitalCardServiceErrorCodes;
+import io.mosip.digitalcard.controller.DigitalCardController;
+import io.mosip.digitalcard.dto.CredentialRequestDto;
+import io.mosip.digitalcard.dto.CredentialResponse;
+import io.mosip.digitalcard.entity.DigitalCardTransactionEntity;
+import io.mosip.digitalcard.exception.DigitalCardServiceException;
+import io.mosip.digitalcard.repositories.DigitalCardTransactionRepository;
+import io.mosip.digitalcard.service.DigitalCardService;
+import io.mosip.digitalcard.service.PDFCardService;
+import io.mosip.digitalcard.util.CredentialUtil;
+import io.mosip.digitalcard.util.DigitalCardRepoLogger;
+import io.mosip.digitalcard.util.RestClient;
+import io.mosip.digitalcard.util.Utility;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.websub.model.EventModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+
+/**
+ * The DigitalCardServiceImpl.
+ *
+ * @author Dhanendra
+ */
+@Service
+public class DigitalCardServiceImpl implements DigitalCardService {
+
+    @Autowired
+    private PDFCardService pdfCardServiceImpl;
+
+    @Value("${mosip.digitalcard.credential.request.partner.id}")
+    private String partnerId;
+
+    @Value("${mosip.digitalcard.credential.type}")
+    private String credentialType;
+
+    @Value("${mosip.digitalcard.fixedRate.in.milliseconds}")
+    private int delaySeconds;
+
+    @Autowired
+    private CredentialUtil credentialUtil;
+
+    @Autowired
+    Utility utility;
+
+    @Autowired
+    RestClient restClient;
+
+    @Autowired
+    DigitalCardTransactionRepository digitalCardTransactionRepository;
+
+    String credentialRequestId=null;
+
+    Logger logger = DigitalCardRepoLogger.getLogger(DigitalCardController.class);
+
+    @Override
+    public boolean generateDigitalCard(EventModel eventModel) {
```

Comment:
We are directly taking in the event model and passing it along to many internal layers, since it is very generic structure, I think the conversion to specific details should be done by the controller and pass the exact details to internal layers. This will also help when there is a change in the specific event structure.

Reply:
if we extract the attribute in the controller layer and send those attributes to the service layer it's not a good practise becoz again we need to extract 4 attribute if something went wrong while extraction then will not get proper error msg also

Reply:
I din't fully get this, but you can extract, validate and then only pass it to the other layers

Reply:
taken care

Reply:
We have to do this for "initiateCredentialRequest(EventModel eventModel);" also

Reply:
taken care